codes/c/chapter_hashing/hash_map_chaining.c

```diff
@@ -0,0 +1,235 @@
+/**
+ * File: hash_map_chaining.c
+ * Created Time: 2023-08-29
+ * Author: SenMing (1206575349@qq.com)
+ */
+
+#include "../utils/common.h"
+
+/* 键值对 int->string */
+typedef struct node {
+    int key;
+    char* val;
+    struct node* next;              // 下一个元素
+} Node;
+
+typedef struct pair {
+    Node* node;                     // 键值对节点
+} Pair;
+
+/* 基于数组简易实现的哈希表 */
+typedef struct ArrayHashMap  {
+    Pair* buckets;                  // 桶数组
+    int size;                       // 键值对数量
+    int capacity;                   // 哈希表容量
+    int extendRatio;                // 扩容倍数
+    double loadThres;               // 触发扩容的负载因子阈值
+} ArrayHashMap;
+
+// 函数声明
+void extend(ArrayHashMap* hashmap);
+
+/* 初始化桶数组 */
+ArrayHashMap* InitArrayHashMap(int tableSize) {
```

Comment:
`newArrayHashMap`

It is better to be consistent with other languages.
For example, initialization functions (constructors in other languages) use `newStructName` , and deinitialization functions (destructors in other languages) use `delStructName`.